roc-wrapper.go

```diff
@@ -0,0 +1,83 @@
+package roc
+
+import "fmt"
+
+func convertErr(err uint32, errstr string) error {
+	if err == 0 {
+		return nil
+	}
+	return fmt.Errorf("%s: %d", errstr, err)
+}
+
+// Init parses the `ip` and `port` and initializes the Address object
+func NewAddress(family Family, ip string, port int) (*Address, error) {
+	var a *Address
+	err := AddressInit(a, family, ip, port)
+	return a, convertErr(err, "Error when initializing an address")
+}
+
+// Family get Address family
+func (a *Address) Family() *Family {
+	return AddressFamily(a)
+}
+
+// Ip get Address ip
+func (a *Address) Ip() *Ip {
+	return AddressIp(a)
+}
+
+// Port get Address port
+func (a *Address) Port() *Port {
+	return AddressPort(a)
+}
+
+func OpenContext(config *ContextConfig) *Context {
```

Comment:
I suggest to make it `OpenContext(config *ContextConfig) (*Context, error)` for convenience (as well as other Open* funcs).

Reply:
I don't think OpenContext returns an error. I think it just returns a pointer and you'll need to check if it's null.

Reply:
> I don't think OpenContext returns an error.

Exactly. I'm suggesting to add the error artificially, like:

```
func OpenContext(config *ContextConfig) (*Context, error) {
    ctx := ContextOpen(config)
    if ctx == nil {
        return nil, errors.New(...)
    }
    return ctx. nil
}
```

This may look a bit stupid from the implementation point of view, but this will give the API user a hint that the function can fail and errors should be checked. Otherwise it may seem like it can't fail since there is no error in the signature, which so typical in the Go world.

Reply:
#3